internal/pkg/cli/task_run.go

```diff
@@ -57,6 +74,12 @@ type runTaskOpts struct {
 	store  store
 	parser dockerfileParser
 	sel    appEnvWithNoneSelector
+
+	docker dockerService
+	ecr    ecrService
+	ec2    vpcService
+	ecs    *ecs.ECS
+	cfn    cloudformation.CloudFormation
```

Comment:
Can you avoid using "ec2", "ecs" as field names for this struct, since they'll be easily confused with the pkg name? Maybe `ecrGetter` etc.

Also could you create/use interface for `ecs` instead of the struct `ecs.ECS`? Same for `cloudformation. CloudFormation`